be/src/udf/java/java_data_converter.cpp

```diff
@@ -0,0 +1,30 @@
+#include "udf/java/java_data_converter.h"
+
+namespace starrocks::vectorized {
+jobject JavaDataTypeConverter::convert_to_object_array(uint8_t** data, size_t offset, int num_rows) {
+    auto& helper = JVMFunctionHelper::getInstance();
+    auto* env = helper.getEnv();
+    jobjectArray arr = env->NewObjectArray(num_rows, helper.object_class(), nullptr);
+    for (int i = 0; i < num_rows; ++i) {
+        env->SetObjectArrayElement(arr, i, reinterpret_cast<JavaUDAFState*>(data[i] + offset)->handle);
+    }
+    return arr;
+}
+
+void JavaDataTypeConverter::convert_to_boxed_array(FunctionContext* ctx, std::vector<DirectByteBuffer>* buffers,
+                                                   const Column** columns, int num_cols, int num_rows,
+                                                   std::vector<jobject>* res) {
+    auto& helper = JVMFunctionHelper::getInstance();
+    ConvertDirectBufferVistor vistor(*buffers);
+    PrimitiveType types[num_cols];
+    for (int i = 0; i < num_cols; ++i) {
+        types[i] = ctx->get_arg_type(i)->type;
+        int buffers_idx = buffers->size();
```

Comment:
suggest to use offset or start instead of buffers_idx

Reply:
fixed